lib/queue.js

```diff
@@ -82,7 +82,7 @@ var Queue = function Queue(name, url, opts){
   }
 
   if(_.isString(url)){
-    opts = _.defaults(redisOptsFromUrl, opts);
+    opts = {redis: _.defaults(redisOptsFromUrl(url), opts)};
```

Comment:
I don't think this is right - it merges opts into the redis options, which is not what the user expects. I think it should be like https://github.com/OptimalBits/bull/pull/538/files#diff-b68fbb0f945cdf9cddec65ddf0fbe456R85 where we only override opts.redis and merge in the user-passed options.